src/xrpld/app/tx/detail/Payment.cpp

```diff
@@ -576,7 +599,16 @@ Payment::doApply()
 
                 if (dstAmount > dstReserve ||
                     sleDst->getFieldAmount(sfBalance) > dstReserve)
-                    return tecNO_PERMISSION;
+                {
+                    if (!credentialsPresent)
```

Comment:
Should this fail regardless of whether credentials are present or not or am I misreading the comments above?

Reply:
If payment is small enough it will proceed despite the DepositPreauth checks.  But if payment is big and we fail with old DepositPreauth we will try new DepositPreauth